fission-core/library/Fission/DataRoot/Class.hs

```diff
@@ -0,0 +1,14 @@
+module Fission.DataRoot.Class
+  ( MonadDataRoot(..)
+  )
+where
+
+import           Fission.Prelude
+import           Servant
+
+import           Fission.AWS.Route53
+import           Fission.User.Username.Types
+import           Network.IPFS.CID.Types
+
+class MonadRoute53 m => MonadDataRoot m where
+  get :: Username -> m (Either ServerError CID)
```

Comment:
`ServerError`, not `ClientError`?

Reply:
Super minor tip, no need to change:

The class definition doesn’t necessarily need to have `MonadRoute53` on it! You _can_ rely on Route53 in the `instance MonadDataRoot Fission`, however, since `Fission` has that defined.

By not requiring `MonadRoute53`, we can later move this lookup to IPNS, the database, or similar. We also don’t need to define `MonadRoute53` if we just want to test the data root in a testing scenario.

Reply:
> The class definition doesn’t necessarily need to have MonadRoute53 on it! You can rely on Route53 in the instance MonadDataRoot Fission, however, since Fission has that defined.

Nice! Yeah that's great!

Reply:
I did `ServerError` here because that is what MonadRoute53 returns in it's other methods (`set` & `clear`)

I can see that it would make sense for those to be Client Errors until they get to the route logic which then returns them as a `ServerError`

Reply:
Ah, okay, that’s our doing in `Fission.AWS.validate`. We use `toServerError` (a creation of our own) to turn the AWS error into something we can pass back to the user.

Probably should do this later in the stack. When we move to MonadRescue, that could be a good idea.

But yeah, reads a bit strange right now but it does the correct things, so 🤷‍♀️